pkg/execution/state/state.go

```diff
@@ -130,7 +143,19 @@ type PauseMutater interface {
 	// leasing or locking mechanism.
```

Comment:
The above comment mentions
> This basic interface does not provide a mechanism for "leasing" a pause temporarily whilst a step is scheduled

But now we have a method to allow it to implemented, should we adjust that comment? I may be missing something